### PR TITLE
Add FusedEmbeddingSeqPoolGradKernel op for merge

### PR DESCRIPTION
**Pull request by zhaify**
- only support on Linux with Intel-MKL enabled
- add padding_idx support
- only support updating words' table with dense matrix
- add UT for FW and BW both on with padding_idx or not

*TODO*

- remove  `d_table->Resize(table_dim);` in line 204

### CODE REVIEW

**Comment by zhaify** [comment]
@yihuaxu please help to review this PR

File: paddle/fluid/operators/fused/fused_embedding_seq_pool_op.h [hunk 1]
```diff
@@ -180,6 +186,59 @@ class FusedEmbeddingSeqPoolGradKernel : public framework::OpKernel<T> {
   void Compute(const framework::ExecutionContext &context) const override {
     auto *table_var = context.InputVar("W");
     DDim table_dim;
+
+#if defined(PADDLE_WITH_MKLML) && !defined(_WIN32) && !defined(__APPLE__) && \
+    !defined(__OSX__) && !defined(PADDLE_WITH_CUDA)
```
Comment (ghost) [inline paddle/fluid/operators/fused/fused_embedding_seq_pool_op.h]: Why "!defined(PADDLE_WITH_CUDA)"? In CI, PADDLE_WITH_CUDA is always defined together with PADDLE_WITH_MKLML. If we have this condition, it means the optimized code here will NEVER be tested in CI testing.

File: paddle/fluid/operators/fused/fused_embedding_seq_pool_op.h [hunk 2]
```diff
@@ -180,6 +186,59 @@ class FusedEmbeddingSeqPoolGradKernel : public framework::OpKernel<T> {
...
+      PADDLE_THROW(
+          "The parameter W of a FusedEmbeddingSeqPoolGradKernel "
+          "must be either LoDTensor");
+    }
+    auto *ids = context.Input<LoDTensor>("Ids");
+    auto *d_output = context.Input<LoDTensor>(framework::GradVarName("Out"));
+    auto *d_table = context.Output<LoDTensor>(framework::GradVarName("W"));
+    int64_t padding_idx = context.Attr<int64_t>("padding_idx");
+
+    d_table->Resize(table_dim);
```
Comment (ghost) [inline paddle/fluid/operators/fused/fused_embedding_seq_pool_op.h:199]: The shape of output tensor "d_table" should be already defined in "[infer_shape](https://github.com/PaddlePaddle/Paddle/blob/3a37c3c7e85763560e07775ef46c523cb318c257/paddle/fluid/operators/fused/fused_embedding_seq_pool_op.cc#L122)". Why we need set here (via Resize) again?

File: paddle/fluid/operators/fused/fused_embedding_seq_pool_op.h [hunk 3]
```diff
@@ -180,6 +186,59 @@ class FusedEmbeddingSeqPoolGradKernel : public framework::OpKernel<T> {
...
+    auto *d_table_data = d_table->mutable_data<T>(context.GetPlace());
+    memset(d_table_data, 0, d_table->numel() * sizeof(T));
```
Comment (ghost) [inline paddle/fluid/operators/fused/fused_embedding_seq_pool_op.h:201]: d_table_data will be the output of sparse GEMM. Do we really need memset it in advance?

**Review by ghost — CHANGES_REQUESTED** [review]

File: paddle/fluid/operators/fused/fused_embedding_seq_pool_op.h (hunk 1 above)
Comment (zhaify) [reply paddle/fluid/operators/fused/fused_embedding_seq_pool_op.h]: done

File: paddle/fluid/operators/fused/fused_embedding_seq_pool_op.h (hunk 2 above)
Comment (zhaify) [reply paddle/fluid/operators/fused/fused_embedding_seq_pool_op.h:199]: done

File: paddle/fluid/operators/fused/fused_embedding_seq_pool_op.h (hunk 3 above)
Comment (zhaify) [reply paddle/fluid/operators/fused/fused_embedding_seq_pool_op.h:201]: done

**Comment by luotao1** [comment]
close due to #19770

**Pull request closed without merging**